src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MAXIMUM_CAPACITY = 1 << 30;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int size;
 
     @Override
     public void put(K key, V value) {
+        if (table == null || table.length == 0) {
+            table = resize();
+        }
+
+        int hash = keyCheck(key);
+
+        int index = hash & (table.length - 1);
 
+        for (Node<K, V> e = table[index]; e != null; e = e.next) {
+            if ((e.hash == hash) && ((e.key == null && key == null)
+                    || (e.key != null && e.key.equals(key)))) {
+                e.value = value;
+                return;
+            }
+        }
+
+        Node<K, V> newNode = table[index];
+        table[index] = new Node<>(hash, key, value, newNode);
+        size++;
+        if (size > threshold) {
+            resize();
+        }
     }
 
-    @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
+
+        int hash = keyCheck(key);
+
+        int index = hash & (table.length - 1);
+
```

Comment:
Also, too much of empty lines across the whole solution, these here are definitely redundant